cola/widgets/startup.py

```diff
@@ -265,16 +336,397 @@ def handle_broken_repo(self, index):
+        self.model().itemChanged.connect(self.item_changed)
+
+        self.action_group = utils.Group(
+            self.open_action,
+            self.open_new_action,
+            self.copy_action,
+            self.launch_editor_action,
+            self.launch_terminal_action,
+            self.open_default_action,
+            self.rename_repo_action,
+            self.delete_recent_action,
+            self.delete_bookmark_action,
+        )
+        self.action_group.setEnabled(True)
+        self.set_default_repo_action.setEnabled(True)
+        self.clear_default_repo_action.setEnabled(True)
+
+    def _is_bookmark_mode(self):
+        return self.view_mode == LIST_VIEW_BOOKMARK
+
+    def _is_recent_mode(self):
+        return self.view_mode == LIST_VIEW_RECENT
+
+    def change_rename_enabled(self, is_enabled):
+        self.rename_repo_action.setEnabled(is_enabled)
+
+    def selected_item(self):
+        index = self.currentIndex()
+        return self.model().itemFromIndex(index)
+
+    def refresh(self):
+        self.model().layoutChanged.emit()
+        context = self.context
+        settings = context.settings
+        builder = BuildItem(context)
+        normalize = display.normalize_path
+
+        entries = [
+            (LIST_VIEW_BOOKMARK, settings.bookmarks),
+            (LIST_VIEW_RECENT, settings.recent),
+        ]
+
+        for mode, entry in entries:
+            items = []
+
+            item = QtGui.QStandardItem(N_('Open...'))
+            item.setEditable(False)
+            item.setIcon(icons.open_directory())
+            items.append(item)
+
+            added = set()
+            for repo in entry:
+                path = normalize(repo['path'])
+                name = normalize(repo['name'])
+                if path in added:
+                    continue
+                added.add(path)
+
+                item = builder.get(path, name)
+                items.append(item)
+
+            self.set_model(items, mode)
+
+    def contextMenuEvent(self, event):
+        """
+        Configures prompt's context menu.
+
+        If self.view_mode is
+            - LIST_VIEW_BOOKMARK, which means this class is for bookmark repos list,
+              context menu DOES NOT contain "Delete from Recent List" action.
+            - LIST_VIEW_RECENT, which means this class is for recent repos list,
+              context menu DOES NOT contain "Delete from Recent List" action.
+        """
+        item = self.selected_item()
+        context = self.context
+        settings = context.settings
+        normalize = display.normalize_path
+
+        paths = set([])
+        if self._is_bookmark_mode():
+            for bookmark in settings.bookmarks:
+                paths.add(normalize(bookmark['path']))
+        elif self._is_recent_mode():
+            for recent in settings.recent:
+                paths.add(normalize(recent['path']))
+
+        if isinstance(item, PromptWidgetItem):
```

Comment:
This is because each view("bookmarks" and "recent") has `Open...` item, and it is `QStandardItem`, not `PromptWidgetItem`. Not to open context menu with `Open...` item click, I added `isinstance` filter.
 
> Meta: we have this weird special Open... folder entry that behaves unlike the rest of them in the UI.
> 
> I almost think we should remove that special entry and rename the Open... button on the bottom left to Browse... for clarity. I suspect that might make the code a little simpler as well.

As following this comment, removing `Open...` item and all `isinstance(item, PromptWidgetItem)` filters from code seems good. How about this?